File: q46_sell_all_cars/q46.py
# Codechef  :  CARSELL
# https://www.codechef.com/problems/CARSELL
# Sell All the Cars

test = int(input())
for tes in range(test) :
    n = int(input())
    arr = list(map(int, input().strip().split(' ')))
    arr.sort()
    j=n-1
    summ = 0
    for i in range(n) :
        numm = arr[i] - j
        if numm > 0 :
            summ = ((numm + summ)%1000000007)
        j-=1
    print(summ)

File: questions/q46_sell_all_cars/q46.cpp
// Codechef  :  CARSELL

// Did not pass

#include<iostream>
using namespace std;

int main()
{
    long long test_cases, tes, n, i, j, k, p, sum, num;
    cin >> test_cases;
    long long *arr;
    for(tes=0;tes<test_cases;tes++)
    {
        cin>>n;
        arr = new long long[n];
        for(i=0;i<n;i++)
        {
            cin>>p;
            if(i==0)
            {
                arr[i] = p;
            }
            else if(p >= arr[i-1])
            {
                arr[i] = p;
            }
            else
            {
                for(j=0;j<i;j++)
                {
                    if(arr[j] > p)
                    {
                        for(k=j;k<i;k++)
                        {
                            arr[k+1] = arr[k];
                        }
                        arr[j] = p;
                    }
                }
            }
        }
        sum = 0;
        p = 0;
        for(i=0;i<n;i++)
        {
            if(arr[i] >= (n-i-1))
            {
                num = arr[i] - (n-i-1);
                sum = (sum+num%1000000007)%1000000007;
            }
            cout<<sum<<" ";
        }
        cout<<sum<<endl;
    }
}